src/com/jetbrains/SystemUtils.java

```diff
@@ -28,4 +28,14 @@ public interface SystemUtils {
      * Implemented for G1 only; equivalent to System.gc() for other garbage collectors.
      */
     void fullGC();
+
+    /**
+     * Performs garbage collection making an additional effort to shrink the heap in size.
+     * After GC the heap size is set proportional to heap used bytes using
+     * -XX:JbrShrinkingGcMaxHeapFreeRatio option.
+     * Implemented for G1 only; equivalent to System.gc() for other garbage collectors.
+     * Equivalent to System.gc() if -XX:JbrShrinkingGcMaxHeapFreeRatio option is not set.
+     */
+    @Extension(Extensions.SHRINKING_GC)
```

Comment:
What is the purpose of `Extensions.SHRINKING_GC`? I don't think it is checked anywhere in the implementation, which, if I understand it right, the extension is always supported.